File: backend/TirthSeva.API/Models/Booking.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TirthSeva.API.Models
{
    public class Booking
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [ForeignKey("User")]
        public int UserId { get; set; }

        [Required]
        [ForeignKey("Temple")]
        public int TempleId { get; set; }

        [ForeignKey("Bhaktnivas")]
        public int? BhaktnivasId { get; set; }

        [ForeignKey("DarshanSlot")]
        public int? DarshanSlotId { get; set; }

        [Required]
        public DateTime BookingDate { get; set; } = DateTime.UtcNow;

        public DateTime? CheckInDate { get; set; }

        public DateTime? CheckOutDate { get; set; }

        [Required]
        [Range(0, 100000)]
        public decimal TotalAmount { get; set; }

        [Required]
        [StringLength(20)]
        public string PaymentStatus { get; set; } = "Pending"; // Pending, Completed, Failed, Refunded

        [Required]
        [StringLength(20)]
        public string BookingStatus { get; set; } = "Confirmed"; // Confirmed, Cancelled, Completed

        [StringLength(500)]
        public string? SpecialRequests { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        public virtual User User { get; set; } = null!;
        public virtual Temple Temple { get; set; } = null!;
        public virtual Bhaktnivas? Bhaktnivas { get; set; }
        public virtual DarshanSlot? DarshanSlot { get; set; }
        public virtual Payment? Payment { get; set; }
    }
}

File: backend-spring/src/main/java/com/tirthseva/api/entity/Booking.java
package com.tirthseva.api.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "Bookings", indexes = {
        @Index(name = "IX_Bookings_UserId", columnList = "userId"),
        @Index(name = "IX_Bookings_BookingDate", columnList = "bookingDate")
})
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Booking {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "Id")
    private Integer id;

    @NotNull
    @Column(name = "UserId", nullable = false)
    private Integer userId;

    @NotNull
    @Column(name = "TempleId", nullable = false)
    private Integer templeId;

    @Column(name = "BhaktnivasId")
    private Integer bhaktnivasId;

    @Column(name = "DarshanSlotId")
    private Integer darshanSlotId;

    @NotNull
    @Column(name = "BookingDate", nullable = false)
    @Builder.Default
    private LocalDateTime bookingDate = LocalDateTime.now();

    @Column(name = "CheckInDate")
    private LocalDate checkInDate;

    @Column(name = "CheckOutDate")
    private LocalDate checkOutDate;

    @NotNull
    @DecimalMin("0")
    @DecimalMax("100000")
    @Column(name = "TotalAmount", nullable = false, precision = 10, scale = 2)
    private BigDecimal totalAmount;

    @NotBlank
    @Size(max = 20)
    @Column(name = "PaymentStatus", nullable = false, length = 20)
    @Builder.Default
    private String paymentStatus = "Pending"; // Pending, Completed, Failed, Refunded

    @NotBlank
    @Size(max = 20)
    @Column(name = "BookingStatus", nullable = false, length = 20)
    @Builder.Default
    private String bookingStatus = "Confirmed"; // Confirmed, Cancelled, Completed

    @NotNull
    @Min(1)
    @Max(1000)
    @Column(name = "NumberOfPersons", nullable = false)
    @Builder.Default
    private Integer numberOfPersons = 1;

    @Size(max = 500)
    @Column(name = "SpecialRequests", length = 500)
    private String specialRequests;

    @Column(name = "CreatedAt", nullable = false)
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    // Navigation properties
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "UserId", insertable = false, updatable = false)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "TempleId", insertable = false, updatable = false)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private Temple temple;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "BhaktnivasId", insertable = false, updatable = false)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private Bhaktnivas bhaktnivas;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "DarshanSlotId", insertable = false, updatable = false)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private DarshanSlot darshanSlot;

    @OneToOne(mappedBy = "booking", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private Payment payment;
}
